lib/request-fix.js

```diff
@@ -0,0 +1,49 @@
+'use strict'
+
+const {Ci} = require('chrome')
+const events = require('sdk/system/events')
+const gw = require('./gateways.js')
+
+/*
+const catman = Cc['@mozilla.org/categorymanager;1'].getService(CI.nsICategoryManager)
+// category ("net-channel-event-sinks")
+catman.addCategoryEntry(in string aCategory, in string aEntry, in string aValue, in boolean aPersist, in boolean aReplace)
+*/
+
+function fixupRequest (e) {
```

Comment:
@the8472  I am going over the backlog and trying to have a better understanding of this part of code.  I see this is related to [protocols.js#L121-L131](https://github.com/the8472/ipfs-firefox-addon/blob/0b085707442b45333e8adef73aa1c3728d0622ba/lib/protocols.js#L121-L131) but can't infer full picture from code alone.

Could you elaborate what is the purpose of this fixup?